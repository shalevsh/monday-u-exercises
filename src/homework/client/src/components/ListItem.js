import React, { useState, useEffect, useCallback } from "react";
import todoService from "../services/todo";
import moment from "moment";
import PropTypes from "prop-types";


function ListItem({ data,deleteItemAction,updateCheckBoxAction}) {
	console.log(data,"data lantsman")
	const [status, setStatus] = useState(data.status);
	const [classes, setClasses] = useState("");
	

	useEffect(() => {
		setClasses(
			`${moment(data.updatedAt).diff(moment(), "minute") < 0 ? "markDeadline":""}
			 ${status ? "checked" : ""}`
		);

	}, [classes]);

	const hanldeUpdateDeadline = e => {
		let date = e.target.value;
		todoService
			.updateDate({
				id: data.id,
				date: date
			})
			.then(() => {
				// reload();
			});
	};
	const hanldeUpdateClick = useCallback(()=>{
		updateCheckBoxAction(data);
	},[updateCheckBoxAction]);


	const hanldeUpdateStatus = e => {
		e.preventDefault();
		if(!status){
		setClasses("checked");
		}else{
		setClasses("");
		}
		setStatus(!status);
	};
	
	const handleDelete = e => {
		e.preventDefault();
		deleteItemAction(data.id);
		
	};
  

	return (
		<>
			<li className={classes}>
				<input
					type="datetime-local"
					className="calendarIcon"
					onChange={hanldeUpdateDeadline}
				/>
				<div>
					<input
						type="checkbox"
						onClick={hanldeUpdateClick}
						checked={status}
						onChange={hanldeUpdateStatus}
					/>
					 {data.item}
				</div>
				<span className="close" onClick={handleDelete}>
					{"\u00D7"}
				</span>
			</li>
		</>
	);
}

ListItem.propTypes = {
	data: PropTypes.object,
	reload: PropTypes.func
};

export default ListItem;
