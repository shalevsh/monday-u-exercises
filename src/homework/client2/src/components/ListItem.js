import React, { useState, useEffect } from "react";
import todoService from "../services/todo";
import moment from "moment";
import PropTypes from "prop-types";

function ListItem({ data, reload }) {
	const [status, setStatus] = useState(data.status);
	const [classes, setClasses] = useState("");

	useEffect(() => {
		setClasses(
			`${
				moment(data.updatedAt).diff(moment(), "minute") < 0
					? "markDeadline"
					: ""
			} ${status ? "checked" : ""}`
		);
	}, []);

	const hanldeUpdateDeadline = e => {
		let date = e.target.value;
		todoService
			.updateDate({
				id: data.id,
				date: date
			})
			.then(() => {
				reload();
			});
	};
	const hanldeUpdateStatus = e => {
		e.preventDefault();
		let n = e.target.checked;
		setStatus(n);
		todoService
			.updateStatus({
				id: data.id,
				status: n
			})
			.then(() => {
				reload();
			});
	};
	const hanldeUpdateStatusLi = e => {
		let n = !status;
		setStatus(n);
		todoService
			.updateStatus({
				id: data.id,
				status: n
			})
			.then(() => {
				reload();
			});
	};

	const handleDelete = e => {
		e.preventDefault();
		todoService.remove(data.id).then(() => {
			const spinning = [
				{ transform: "rotate(0) scale(1)" },
				{ transform: "rotate(0deg) scale(0)" }
			];
			const timing = {
				duration: 1000,
				iterations: 1
			};
			e.target.parentNode.animate(spinning, timing);
			setTimeout(() => reload(), 1000);
		});
	};

	return (
		<>
			<li className={classes}>
				<input
					type="datetime-local"
					className="calendarIcon"
					onChange={hanldeUpdateDeadline}
				/>
				<div onClick={hanldeUpdateStatusLi}>
					<input
						type="checkbox"
						checked={status}
						onChange={hanldeUpdateStatus}
					/>
					{data.item}
				</div>
				<span className="close" onClick={handleDelete}>
					{"\u00D7"}
				</span>
			</li>
		</>
	);
}

ListItem.propTypes = {
	data: PropTypes.object,
	reload: PropTypes.func
};

export default ListItem;
