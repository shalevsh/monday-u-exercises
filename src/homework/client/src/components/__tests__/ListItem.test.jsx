import renderer from "react-test-renderer";
import ListItem  from "../ListItem"

 const item={
            id: 1,
            isDisplay: false,
            isPokemon: false,
            item: "four",
            status: false
        };
    
test("renders correctly pending todo", () => {
  const tree = renderer
    .create(<ListItem data={item}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});