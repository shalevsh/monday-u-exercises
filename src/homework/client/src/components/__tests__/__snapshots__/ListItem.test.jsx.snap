// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly pending todo 1`] = `
<li
  className=""
>
  <input
    className="calendarIcon"
    onChange={[Function]}
    type="datetime-local"
  />
  <div>
    <input
      checked={false}
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    four
  </div>
  <span
    className="close"
    onClick={[Function]}
  >
    Ã—
  </span>
</li>
`;
